app/core/circuit_breaker.py

```
@@ -0,0 +1,217 @@
+        self.name = name
+        self.failure_threshold = failure_threshold
+        self.recovery_timeout = recovery_timeout
+        self.half_open_timeout = half_open_timeout
+        self.reset_timeout = reset_timeout
+        
+        self._state = CircuitState.CLOSED
+        self._stats = CircuitStats()
+        self._last_state_change = time.time()
+        self._lock = asyncio.Lock()
+    
+    async def execute(self, func: Callable[..., T], *args, **kwargs) -> T:
```

Ruff (0.8.2)

Missing type annotation for `*args`

(ANN002)